### ISSUE TITLE
Wrong size of the dmg window

### ISSUE THREAD

**Issue by Uznick** [create-dmg/create-dmg#20]
I create the dmg using command:

`yoursway-create-dmg/create-dmg --window-size 415 295 --background ~/Downloads/background.png --volicon ~/Downloads/.VolumeIcon.icns --icon-size 129 --volname "Firefox" --app-drop-link 320 152 --icon "Firefox" 95 152 firefox_my.dmg Firefox.app`

The background.png is 415x295. When the .dmg file is created, I get the white bar to the right of the background, and the size of the window I get is greater than 415 px: http://cl.ly/image/2z2i3Z050s1p

<img width="439" alt="Screen Shot 2020-05-28 at 8 47 32 AM" src="https://user-images.githubusercontent.com/2618447/83143403-e6d2e500-a0bf-11ea-9b71-be555bbafb9c.png">

How can I fix that? I've tried to play with the --window-size, but it did not help.

**Comment by eraxillan** [comment]
Hi!
So, still no fix/workaround for this issue?

P.S. I'm able to reproduce this bug only on OS X Mountain Lion (10.8).

**Comment by azadi** [comment]
@eraxillan, @Uznick: I had the same issue. Make sure the PNG image you use for the background has a resolution/print size of 72 DPI. I was using Inkscape to export the PNG which defaults to 90. I then edited the image in GIMP, changed the print size (Image > Print Size) to 72 and then it finally worked.

**Comment by zaygraveyard** [comment]
After some trial and error, I found that (as @azadi mentioned) DPI matters.
It looks like the actual width displayed is `displayed-width = image-width * 72 / image-dpi-width` and same for the height.
I used `sips` to set the DPI to 144 (for retina displays):
```bash
sips --setProperty dpiWidth 144 --setProperty dpiHeight 144 backgroung.png
```
Hope this helps.

**Comment by aonez** [comment]
@zaygraveyard does your background have 72 and 144 images, a TIFF containing both retina/non-retina resolutions?

**Comment by zaygraveyard** [comment]
@aonez No, it's a PNG with 144 DPI and it works on both retina and non-retina

**Comment by aonez** [comment]
@zaygraveyard I'm just doing some quick tests. `sips --getProperty dpiHeight` seems to return 72 for a 72 PNG and 144 for a 144 PNG.
What was your PNG returning before executing the  `--setProperty`? Can you share the PNG (or some sample) and maybe a DMG sample with that 144?

I use `tiffutil -cathidpicheck "background.png" "background@2x.png" -out "background.tiff"` two merge both retina and no retina images in a TIFF, but if using a properly set 144 is enough, will save some space.

**Comment by zaygraveyard** [comment]
@aonez I don't understand the problem that you have with using a single 144 DPI PNG
That what I'm using and I found that it works well on both retina and non-retina displays (given it's big enough)
The original DPI of the image doesn't matter

**Comment by aonez** [comment]
Don't read me wrong @zaygraveyard, no problem at all, I'm just trying to understand the DPI think to enhance the `create-dmg` script. So for example I've read that @azadi's image was 90 DPI, it can be detected with `sips` and fixed to 72. I was curious what you 144 DPI image was reading from `sips` and why setting it to 144 fixed the issue.

**Comment by zaygraveyard** [comment]
@aonez Oh I was using Inkscape to create the background.

When exporting, Inkscape uses 90 DPI, even when changing the size so when I doubled the size the DPI doubled also.

AFAIK the DPI property in an image is purely informative, I guess Inkscape considers screens to have 90 DPI (or maybe it's for print)

Since Finder expects the DPI to be 72 (`size-in-points = size-in-pixels * 72 / dpi`), if the image has a DPI of 90 then it will be displayed 20% smaller.

So I just correct the DPI using `sips` :smiley: 

I would also like to note:
1. I'm using an image with double the size and double the DPI (144) so that it has enough pixels for the retina displays
2. I don't belive `create-dmg` should change the DPI automatically (there might be valid reason for having a different DPI), but it could be an CLI option
3. Your TIFF method can be used in cases where the high resolution image needs to be different  from the low resolusion one

**Comment by zaygraveyard** [comment]
I now use `qlmanage -t` to convert the SVG into a PNG without passing by Inkscape
One downside to this technique is that the resulting PNG will be a square (padded to the right and bottom as needed)
